### DISCUSSION TITLE
Reusing actions on actors of unknown type?

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [bitbrain/beehave#269] [Q&A]
Take a script like this for example...
```
class_name MoveTowardMouse extends ActionLeaf


func tick(actor, blackboard: Blackboard):
	var target_position := get_viewport().get_mouse_position()
	actor.position.move_towards(target_position)
	
	if actor.position.distance_to(target_position) < 1:
		return SUCCESS
	return RUNNING

```

How do we know actor contains a variable `position` of type `Vector2`?
How do we ensure the action is not attached to something it can't work on?

**Comment by lostptr** [comment]
Yeah I miss having this but I dont think is possible due to the nature of gdscript. It works with [duck typing](https://en.wikipedia.org/wiki/Duck_typing#:~:text=Duck%20typing%20is%20similar%20to,that%20is%20accessed%20during%20runtime.) and its very useful for prototyping. For this to be possible we would need gdscript to have [generics](https://github.com/godotengine/godot-proposals/issues/1207) and [interfaces (or traits)](https://github.com/godotengine/godot-proposals/issues/6416). 

PS: Now I see you where the one that opened the generics proposal lmao

**Comment by RedstoneParadox** [comment]
I personally just do this at the start of `tick()`
```gdscript
var _actor : Actor = actor as Actor
```
It's not the most elegant solution and will crash if `actor` is not an `Actor` instance, but I don't intend on attaching a behavior tree to something that isn't an `Actor`.

**Comment by lostptr** [comment]
Could this be a planned feature for #285?
Not that it would work on gdscript, but would there be a possibility of doing this to enable this feature if the person is using, say, C#?
Just a thought tho
